pkg/webhook/webhook.go

```diff
@@ -124,7 +124,7 @@ func getAdmissionResponseForBuild(ar *v1beta1.AdmissionReview, quayIntegration *
 
 	quayRegistryHostname, err := quayIntegration.GetRegistryHostname()
 
-	if buildv1.JenkinsPipelineBuildStrategyType == build.Spec.Strategy.Type || build.Spec.CommonSpec.Output.To == nil || build.Spec.CommonSpec.Output.To.Kind != "ImageStreamTag" {
+	if (build.Spec.Strategy.DockerStrategy == nil && build.Spec.Strategy.SourceStrategy == nil) || build.Spec.CommonSpec.Output.To.Kind != "ImageStreamTag" {
```

Comment:
What about custom build strategies? Other than that, assuming that Docker and S2I strategies mandate that `spec.ouput.to` is not NIL I think this lgtm

Reply:
Custom Build strategies is a big question on whether it should be supported which is why I went down my original approach to really only exclude `JenkinsPipeline` and then do checks on whether certain fields are present.

`Output` is not a pointer so there will not be an issue of hitting a NPE

Reply:
Ack on custom builds not being supported. I think we should not break the system when we encounter such a build. People could be mixing several types of build configs in a namespace. Not supporting it by gracefully ignoring those seems the best way to me.

Reply:
As the logic stands now, custom strategies would not have their output image destinations mutated and would be ignored. Is that the expectation that we would want to deliver?

Reply:
Sounds good to me for now.